### ISSUE TITLE
Configure Autosubmit to work in Athos

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1167]
In GitLab by @nperezzanon on Nov 30, 2023, 12:19

Hello @dbeltrankyl and @kinow,

I am trying to use Autosubmit at ECMWF. I have a conda environment with this autosubmit version:

```yaml
    - autosubmit==4.0.98
    - autosubmitconfigparser==1.0.49
```

I haven't been able to create an expid yet. I am currently trying to define the HOST in the platform.yml file.

To connect to the machine I run the command `tsh login --proxy=jump.ecmwf.int` and then I log in the web page with a secure key and then I run the command `ssh -J cyce@jump.ecmwf.int cyce@hpc-login`

This is my original `.ssh/config` file content:

```bash
Host jump.ecmwf.int a?-* a??-* hpc-* hpc2020-* ecs-*
  User cyce
  IdentityFile ~/.tsh/keys/jump.ecmwf.int/nuria.perez@ bsc.es
  CertificateFile ~/.tsh/keys/jump.ecmwf.int/nuria.perez@ bsc.es-ssh/jump.ecmwf.int-cert.pub
  HostKeyAlgorithms +ssh-rsa*,rsa-sha2-512
  PubkeyAcceptedKeyTypes +ssh-rsa*
  ServerAliveInterval 60
  TCPKeepAlive yes
```

I am trying to modify it to something like this:

```bash
Host athos-hpc
  HostName hpc-login
  User cyce
  IdentityFile ~/.ssh/id_rsa
  ProxyJump jump.ecmwf.int 
```
But I am not succeeding. Do you know what should be the name of the HOST with this info or how should I modify my local .ssh/config so `athos-hpc` becomes the HOST?

My latest attempt has been to try this:

```bash
Host Athos-hpc
  HostName hpc-login
  User cyce
  IdentityFile ~/.tsh/keys/jump.ecmwf.int/nuria.perez@ bsc.es
  CertificateFile ~/.tsh/keys/jump.ecmwf.int/nuria.perez@ bsc.es-ssh/jump.ecmwf.int-cert.pub
  HostKeyAlgorithms +ssh-rsa*,rsa-sha2-512
  PubkeyAcceptedKeyTypes +ssh-rsa*
  ServerAliveInterval 60
  TCPKeepAlive yes
  ProxyJump jump.ecmwf.int
```

But when trying `ssh Athos-hpc` I got the error message: 

```
Permission denied (publickey).
kex_exchange_identification: Connection closed by remote host
```
and when using `tsh Athos-hpc` I got `ERROR: expected command but got "Athos-hpc"`.

Well, let me know your opinion.

Thanks a lot,

Núria

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Nov 30, 2023, 14:06

Hello @nperezzanon , 

First, I would use 4.0.101 or 4.0.92. v4.0.98 has a bug if you use a particular feature ( additional files) if not, you're safe. 

I recommend to following ( omitting the steps of bsc workstations if don't apply): 

https://earth.bsc.es/gitlab/es/auto-ecearth3/-/wikis/Running-on-ECMWF-HPC2020

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Nov 30, 2023, 14:07

If you can't still manage to run it with the help of the information mentioned, you can ask me ( that guide is well-tested )

**Comment by kinow** [comment]
In GitLab by @nperezzanon on Dec 1, 2023, 13:03

Hi @dbeltrankyl 

Thanks a lot for the link. I am able to connect now using `ssh hpc-login`. I have followed also the 'initial set up' steps,  and when I run the command `micromamba activate cmor_3.5_py_2.7`, I get the following error:

```
'micromamba' is running as a subprocess and can't modify the parent shell.
Thus you must initialize your shell before using activate and deactivate.

To initialize the current bash shell, run:
    $ eval "$(micromamba shell hook --shell bash)"
and then activate or deactivate with:
    $ micromamba activate
To automatically initialize all future (bash) shells, run:
    $ micromamba shell init --shell bash --root-prefix=~/micromamba
If your shell was already initialized, reinitialize your shell with:
    $ micromamba shell reinit --shell bash
Otherwise, this may be an issue. In the meantime you can run commands. See:
    $ micromamba run --help

Supported shells are {bash, zsh, csh, xonsh, cmd.exe, powershell, fish}.
critical libmamba Shell not initialized
```
Do you know how to solve it?

Thanks in advance,

Núria

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 1, 2023, 13:07

I'm glad that it helped you!

I'm not familiar with using mamba, conda, or microamba at all, usually, I just use venv or modules @kinow do you have any idea?

**Comment by kinow** [comment]
In GitLab by @nperezzanon on Dec 1, 2023, 13:11

I wonder if I really need this to work. My plan is to run Autosubmit in Athos, not submit the jobs from BSC. Furthermore, I am using a conda environment that already contains autosubmit.

**Comment by kinow** [comment]
Sorry @nperezzanon , I was planning to log in to the ECMWF env to have a look at the settings there and try to suggest some workaround, but I spent the day on another task, and have a final meeting now at 5PM. Tomorrow I will have a few other meetings pre-holiday, then back next week. If you have not yet made progress here I will try to prioritize this one for the beginning of the next week.

**Comment by kinow** [comment]
In GitLab by @nperezzanon on Dec 14, 2023, 15:57

Hi,

@V-Agudetse has been helping me to configure sunset. We had an expid but we got the following error when running the code `nohup.out`:

```
Autosubmit is running with v4.0.98
Checking the connection to all platforms in use
Traceback (most recent call last):
  File "/perm/cyce/conda/envs/condaCerise/lib/python3.11/site-packages/autosubmit/autosubmit.py", line 2045, in run_experiment
    job_list, submitter , exp_history, host , as_conf, platforms_to_test, packages_persistence, _ = Autosubmit.prepare_run(expid, notransitive,start_time, start_after, run_only_members)
                                                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/perm/cyce/conda/envs/condaCerise/lib/python3.11/site-packages/autosubmit/autosubmit.py", line 1985, in prepare_run
    Autosubmit.restore_platforms(platforms_to_test)
  File "/perm/cyce/conda/envs/condaCerise/lib/python3.11/site-packages/autosubmit/autosubmit.py", line 2334, in restore_platforms
    raise AutosubmitCritical("Issues while checking the connectivity of platforms.", 7010, issues + "\n" + ssh_config_issues)
log.log.AutosubmitCritical:

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/perm/cyce/conda/envs/condaCerise/bin/autosubmit", line 37, in main
    Autosubmit.parse_args()
  File "/perm/cyce/conda/envs/condaCerise/lib/python3.11/site-packages/autosubmit/autosubmit.py", line 658, in parse_args
    return Autosubmit.run_experiment(args.expid, args.notransitive,args.start_time,args.start_after, args.run_only_members, args.profile)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/perm/cyce/conda/envs/condaCerise/lib/python3.11/site-packages/autosubmit/autosubmit.py", line 2048, in run_experiment
    raise AutosubmitCritical(e.message, 7014, e.trace)
log.log.AutosubmitCritical:
^[[38;5;214m[ERROR] Trace:
[ATHOS] has configuration issues.
 Check that the connection is passwd-less.(ssh cyce@hpc-login-spesiccf)
 Check the parameters that build the root_path are correct:{scratch_dir/project/user} = {/ec/res4/scratch/cyce/spesiccf/cyce}
Timeout connection this is an PARAMIKO SSHEXCEPTION: indicates that there is something incompatible in the ssh_config for host:hpc-login-spesiccf
 maybe you need to contact your sysadmin^[[0m^[[39m
^[[1m ^[[31m[CRITICAL] Issues while checking the connectivity of platforms. [eCode=7014]^[[0m^[[39m
More info at https://autosubmit.readthedocs.io/en/master/troubleshooting/error-codes.html

```

We have tried to set the run to be local, but there still have issues. Do you know how to configure to run in ATHOS, which is is slurm, but submitting the jobs locally?

Thanks in advance,

Núria

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 15, 2023, 08:22

Hello @nperezzanon ,

Does the `/ec/res4/scratch/cyce/spesiccf/cyce` path exists?

**Comment by kinow** [comment]
In GitLab by @nperezzanon on Dec 15, 2023, 09:39

Hi Dani,

Yes, the path exists. I created following the instructions you shared: https://earth.bsc.es/gitlab/es/auto-ecearth3/-/wikis/Running-on-ECMWF-HPC2020

Núria

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 15, 2023, 09:52

And did you perform the ecaccess-certificate-create command? 

This error happens when Autosubmit can't write a file inside that folder ( to check permissions, connectivity etc ). 

In ecmwf platform, autosubmit uses ecaccess. Make sure that you have the module load and updated before running autosubmit ( or the commands are available where you launch Autosubmit)

**Comment by kinow** [comment]
In GitLab by @nperezzanon on Dec 15, 2023, 10:14

No, I haven't run this command. I thought this was necessary to run Autosubmit from BSC because I don't have the module ecaccess-webtoolkit/6.3.1. 

I have just tried to run `ecaccess-certificate-create` (with the conda environment I have at ECMWF), and I got `ecaccess-certificate-create: command not found `.

Does this means I need more software in the conda environment, right? 

Núria

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 15, 2023, 10:43

Yes, you need the ecaccess software so autosubmit can access these commands in order to communicate with the ecmwf hpc

**Comment by kinow** [comment]
In GitLab by @eferre1 on Dec 19, 2023, 15:05

Summarizing the results of the meeting that we had today (and also adding myself to the issue in case further help is needed):

- Since the installation of autosubmit in the conda environment is located at ECMWF-HPC2020 (ATHOS) machine, `ecaccess` is not required. What was missing is the initial setup on the machine (`ssh-key-setup`) so that an `ssh` can be done internally to any machine of the ECMWF complex. (also, the micromamba commands that were mentioned above are not needed unless running auto-ecearth3). Then, the `HOST` to use is `hpc-batch` and the connection via ssh works.
- With this setup, autosubmit was succeeding in the connectivity check, and then failed due to some other issues with the config:
  -  For some reason (further investigation required), probably some parameter overwriting, the `PROJECT` variable was being added to the `HOST`, resulting in the connectivity check to `hpc-batch-spesiccf` failing.
  - Also, since this is not on the usual auto-ecearth3 ECMWF accounts, the `PROJECT` is not `spesiccf` (the auto-ecearth3 wiki used as reference has that since it is always the same), but in this case it is `copext` (this is a minor change in the settings). But for some reason, setting this as the `PROJECT` in the platforms config file, makes the submission fail due to `wrong qos`. Testing it manually, removing the `#SBATCH -A copext` line from the header made it work, without any `-A` it should use the default account, which checking with the `groups` command, it was only this `copext` group existing.

So, some tweaking is still required to make it work (Nuria and Victoria will keep working on that, and feel free to ask any doubts), but it should be able to be used from this autosubmit installation in a conda environment inside the ECMWF machine.

Apart from that, we also talked about the "issue" of running autosubmit:
- from the autosubmit hubs, as it would facilitate giving support in case of autosubmit issues. But that depends on the project requirements and has to be checked if possible or not (it would require also some setup to make it work, but nothing too complex).
- from the hpc-login node: it needs to be asked to ECMWF if a long running process (autosubmit run) will be killed (since the process needs to keep running to check the status and keep submitting new jobs) or something can be done in regards of having it there (similar to DestinE VM in LUMI). In the final version, from Nuria's comment, it can be up until ~300 jobs, which will be limited to be running ¿20? at a time (I think the maximum mentioned was 100, but in any case, not all would be running simultaneously, so I'm taking a more "realistic" aproach, and still being "a bit optimistic" on the amount of jobs running in parallel), and they can take ~1 hour, which means that at optimal performance, it would take at least 15 hours to run, hence the question about where it should run

**Comment by kinow** [comment]
In GitLab by @nperezzanon on Dec 20, 2023, 10:09

Hi @eferre1 and @kinow 

Thanks a lot for your report and for yesterday's meeting.

I have just checked ECMWF documentation and the 'account' we were using might be incorrect. Following this pagehttps://confluence.ecmwf.int/display/UDOC/HPC2020%3A+Accounting, there is a section 'What are my project accounts?', and I got:

```
[cyce@ac6-101 sunset]$ account 
eccopext
```
I will try it soon and report back here.

Thanks again for your help.

Núria

**Comment by kinow** [comment]
Hi @nperezzanon is this issue still needed, or can we close this? Thanks!
